dask_expr/_collection.py

```diff
@@ -1851,6 +1856,31 @@ def nbytes(self):
     def keys(self):
         return self.index
 
+    def __array_ufunc__(self, numpy_ufunc, method, *inputs, **kwargs):
+        out = kwargs.get("out", ())
+        for x in inputs + out:
+            # ufuncs work with 0-dimensional NumPy ndarrays
+            # so we don't want to raise NotImplemented
+            if isinstance(x, np.ndarray) and x.shape == ():
+                continue
+            elif not isinstance(
+                x, (Number, Scalar, FrameBase, Array, pd.DataFrame, pd.Series, pd.Index)
+            ):
+                return NotImplemented
+
+        if method == "__call__":
+            if numpy_ufunc.signature is not None:
+                return NotImplemented
+            if numpy_ufunc.nout > 1:
+                # ufuncs with multiple output values
+                # are not yet supported for frames
+                return NotImplemented
+            else:
+                return elemwise(numpy_ufunc, *inputs, **kwargs)
+        else:
+            # ufunc methods are not yet supported for frames
+            return NotImplemented
```

Comment:
I am a bit confused why it's a problem to not have this in FrameBase, it's exactly the same method copied to both classes, no?

Reply:
Scalar inherits from FrameBase as well and can't have access to it. There are hasattr checks involved unfortunately